test/WebSites/TempDataWebSite/Controllers/HomeController.cs

```diff
@@ -17,5 +18,30 @@ public IActionResult DisplayTempData(string value)
             TempData["key"] = value;
             return View();
         }
+
+        public IActionResult SetTempDataAndRedirect(string value)
+        {
+            TempData["key"] = value;
+            return RedirectToAction("GetTempData");
+        }
+
+        public string GetTempData()
+        {
+            var value = TempData["key"];
+            return value.ToString();
+        }
+
+        public IActionResult SetTempDataAndRedirectToPeek(string value)
+        {
+            TempData["key"] = value;
+            return RedirectToAction("PeekTempData");
+        }
+
+        public IActionResult PeekTempData()
+        {
+            var peekValue = TempData.Peek("key");
+            var getValue = TempData["key"];
```

Comment:
As discussed, we should have a test that doing a Peek keeps the value for the next request and doing a Get does not.